lib/util.ml

```diff
@@ -224,6 +224,36 @@ let search_in_path p s =
 
 let search_in_assets = search_in_path Secure.assets
 
+let hash_file file =
+  if Sys.file_exists file then
+    try
+      let ic = open_in_bin file in
+      Fun.protect
+        ~finally:(fun () -> close_in ic)
+        (fun () ->
+          let content = really_input_string ic (in_channel_length ic) in
+          Digest.to_hex (Digest.string content))
```

Comment:
It is a bit sad to allocate a huge string and throw it after. Digest computes the hash per blocks, so you can read the channel in a buffer of fixed size and compute incrementally the digest.